lib/remote/apilistener-configsync.cpp

```diff
@@ -393,6 +395,45 @@ void ApiListener::UpdateConfigObject(const ConfigObject::Ptr& object, const Mess
 	}
 }
 
+/**
+ * Syncs the specified object and its direct and indirect parents to the provided client
+ * in topological order of their dependency graph recursively.
+ *
+ * Objects that the client does not have access to are skipped without going through their dependency graph.
+ *
+ * Please do not use this method to forward remote generated cluster updates; it should only be used to
+ * send local updates to that specific non-nullptr client.
+ *
+ * @param object The config object you want to sync.
+ * @param azone The zone of the client you want to send the update to.
+ * @param client The JsonRpc client you send the update to.
+ * @param syncedObjects Used to cache the already synced objects.
+ */
+void ApiListener::UpdateConfigObjectWithParents(const ConfigObject::Ptr& object, const Zone::Ptr& azone,
+	const JsonRpcConnection::Ptr& client, std::unordered_set<ConfigObject*>& syncedObjects)
+{
+	if (syncedObjects.find(object.get()) != syncedObjects.end()) {
+		return;
+	}
+
+	/* don't sync objects for non-matching parent-child zones */
+	if (!azone->CanAccessObject(object)) {
+		return;
+	}
+	syncedObjects.emplace(object.get());
+
+	for (const auto& parent : DependencyGraph::GetParents(object)) {
+		// Actually, the following dynamic cast should never fail, since the DependencyGraph class
+		// expects the types to always be of type Object::Ptr and such an object is supposed to always
+		// point to an instance of the specific derived ConfigObject class. See TypeHelper<>::GetFactory().
+		if (auto parentObj = dynamic_pointer_cast<ConfigObject>(parent)) {
```

Comment:
* Even if you don't approve #10264 as such, which is ok, can you just admit that this code comment literally screams for it?

Reply:
Alternatively, you could use static_cast. Just as an idea.